src/main/java/org/springframework/data/convert/PropertyValueConverterFactories.java

```diff
@@ -0,0 +1,240 @@
+			this.delegates = Collections.unmodifiableList(delegates);
+		}
+
+		@Nullable
+		@Override
+		public <A, B, C extends ValueConversionContext<?>> PropertyValueConverter<A, B, C> getConverter(
+				PersistentProperty<?> property) {
+			return delegates.stream().map(it -> (PropertyValueConverter<A, B, C>) it.getConverter(property))
+					.filter(Objects::nonNull).findFirst().orElse(null);
+		}
+
+		@Override
+		public <S, T, C extends ValueConversionContext<?>> PropertyValueConverter<S, T, C> getConverter(
+				Class<? extends PropertyValueConverter<S, T, C>> converterType) {
+			return delegates.stream().filter(it -> it.getConverter(converterType) != null).findFirst()
+					.map(it -> it.getConverter(converterType)).orElse(null);
+		}
+
+		public List<PropertyValueConverterFactory> converterFactories() {
+			return delegates;
+		}
+	}
+
+	/**
+	 * Trivial implementation of {@link PropertyValueConverterFactory} capable of instantiating a
+	 * {@link PropertyValueConverter} via default constructor or in case of {@link Enum} returning the first value.
+	 *
+	 * @author Christoph Strobl
+	 * @since 2.7
+	 */
+	static class SimplePropertyConverterFactory implements PropertyValueConverterFactory {
+
+		@Override
+		public <S, T, C extends ValueConversionContext<?>> PropertyValueConverter<S, T, C> getConverter(
+				Class<? extends PropertyValueConverter<S, T, C>> converterType) {
+
+			Assert.notNull(converterType, "ConverterType must not be null!");
+
+			if (converterType.isEnum()) {
+				return (PropertyValueConverter<S, T, C>) EnumSet.allOf((Class) converterType).iterator().next();
+			}
+			return BeanUtils.instantiateClass(converterType);
+		}
+	}
+
+	/**
+	 * {@link PropertyValueConverterFactory} implementation that leverages the {@link BeanFactory} to create the desired
+	 * {@link PropertyValueConverter}. This allows the {@link PropertyValueConverter} to make use of DI.
+	 *
+	 * @author Christoph Strobl
+	 * @since 2.7
+	 */
+	static class BeanFactoryAwarePropertyValueConverterFactory implements PropertyValueConverterFactory {
+
+		private final BeanFactory beanFactory;
+
+		BeanFactoryAwarePropertyValueConverterFactory(BeanFactory beanFactory) {
+			this.beanFactory = beanFactory;
+		}
+
+		@Override
+		public <S, T, C extends ValueConversionContext<?>> PropertyValueConverter<S, T, C> getConverter(
+				Class<? extends PropertyValueConverter<S, T, C>> converterType) {
+
+			Assert.state(beanFactory != null, "BeanFactory must not be null. Did you forget to set it!");
+			Assert.notNull(converterType, "ConverterType must not be null!");
+
+			try {
+				return beanFactory.getBean(converterType);
+			} catch (NoSuchBeanDefinitionException exception) {
+
+				if (beanFactory instanceof AutowireCapableBeanFactory) {
+					return (PropertyValueConverter<S, T, C>) ((AutowireCapableBeanFactory) beanFactory).createBean(converterType,
+							AutowireCapableBeanFactory.AUTOWIRE_CONSTRUCTOR, false);
+				}
+			}
+			return null;
```

Comment:
Shouldn't we throw an exception here if we cannot instantiate the converter? A silent fall-through doesn't seem appropriate.